source/common/network/connection_impl.cc

```diff
@@ -687,6 +696,24 @@ ClientConnectionImpl::ClientConnectionImpl(
   }
 }
 
+void ClientConnectionImpl::close(ConnectionCloseType type) {
+  if (type == ConnectionCloseType::ResetNoFlush) {
```

Comment:
What happens if someone calls this on the ServerConnectionImpl?

Reply:
Maybe you mean `ConnectionImpl::close()`? It now starts with
```
ASSERT(type != ConnectionCloseType::ResetNoFlush);
```